File: lib/models/jugador_model.dart
import 'club_model.dart';

class JugadorModel {
  final String id;
  final String nombre;
  final Club club;
  final String categoria; // 'Damas' o 'Caballeros'
  final String division;
  final int numeroCamiseta;
  final String posicion;
  final int goles;
  final int partidos;
  final DateTime fechaNacimiento;
  final String? fotoPath;

  JugadorModel({
    required this.id,
    required this.nombre,
    required this.club,
    required this.categoria,
    required this.division,
    required this.numeroCamiseta,
    required this.posicion,
    required this.goles,
    required this.partidos,
    required this.fechaNacimiento,
    this.fotoPath,
  });
}

File: app/src/main/java/com/example/hockey_app/data/models/JugadorModel.kt
package com.example.hockey_app.data.models

import kotlinx.serialization.Serializable

@Serializable
data class JugadorModel(
    val id: String,
    val nombre: String,
    val club_id: String = "0",
    val club_nombre: String = "Sin Club",
    val club_escudo: String? = null,
    val categoria: String = "Damas",
    val division: String = "1ra",
    val numero_camiseta: Int = 0,
    val posicion: String = "Jugador",
    val goles: Int = 0,
    val partidos: Int = 0,
    val fecha_nacimiento: String? = null,
    val foto_url: String? = null
)
